From: "Steven M. Robbins" <smr@sumost.ca>
Date: Sat, 19 Apr 2014 01:43:25 -0500
Subject: double free corruption

Remove valgrind error "Source and destination overlap in memcpy(0xf121b90, 0xf121b98, 944)".
This is the cause of the "double free or corruption" error seen:
https://github.com/harmattan/numptyphysics/issues/2

Bug: https://bugs.debian.org/749025
Forwarded: https://github.com/harmattan/numptyphysics/pull/3
---
 Array.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Array.h b/Array.h
index 8acd7d4..d84a39f 100644
--- a/Array.h
+++ b/Array.h
@@ -103,7 +103,7 @@ class Array
     if (i >= 0 ) {
       ASSERT( i < m_size );
       if ( i < m_size-1 ) {
-	memcpy( m_data+i, m_data+i+1, (m_size-i-1)*sizeof(T) );
+	memmove( m_data+i, m_data+i+1, (m_size-i-1)*sizeof(T) );
       }
       m_size--;
     }
